test/realm_value_test.dart

```diff
@@ -0,0 +1,158 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+
+import 'package:test/test.dart' hide test, throws;
+import '../lib/realm.dart';
+
+import 'test.dart';
+
+part 'realm_value_test.g.dart';
+
+@RealmModel()
+class _AnythingGoes {
+  @Indexed()
+  late RealmValue oneAny;
+  late List<RealmValue> manyAny;
```

Comment:
```suggestion
late RealmValue oneValue;
late List<RealmValue> manyValues;
```
Maybe will be better if you rename all the places where you have members `any` to `value`. 'Any' is confusing to me.

Reply:
> We probably need to expose `ObjectType` similarly to what we're doing in C#.

I will re-work to disallow non top-level objects by type in RealmValue. This will take a bit of time.

Reply:
Much easier than anticipated. We now prevent storing embedded objects in a `RealmValue` as they implement `EmbeddedObjectMarker` and not `TopLevelObjectMarker`

Reply:
.. hmm these comments are misplaced.. will delete